table/tables/tables.go

```diff
@@ -1190,23 +1194,169 @@ func init() {
 	table.MockTableFromMeta = MockTableFromMeta
 }
 
+// sequenceCommon cache the sequence value.
+// `alter sequence` will invalidate the cached range.
+// `setval` will recompute the start position of cached value.
+type sequenceCommon struct {
+	meta *model.SequenceInfo
+	// base < end when increment > 0.
+	// base > end when increment < 0.
+	end  int64
+	base int64
+	// round is used to count the cycle times.
+	round int64
+	mu    sync.Mutex
+}
+
+// GetSequenceBaseEndRound is used in test.
+func (s *sequenceCommon) GetSequenceBaseEndRound() (int64, int64, int64) {
+	return s.base, s.end, s.round
+}
+
```

Comment:
Please use `RWMutex` and give enough encapsulation to restrict the access for both reading and writing on each field.

Reply:
Addressed